CMakeLists.txt

```diff
@@ -171,6 +171,10 @@ if (WITH_CUDA)
         list(APPEND CUDA_NVCC_FLAGS "-D_MWAITXINTRIN_H_INCLUDED -D_FORCE_INLINES")
       endif()
       SET(CUDA_PROPAGATE_HOST_FLAGS OFF)
+
+      if (CMAKE_OSX_SYSROOT)
+        set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -Xcompiler -isysroot -Xcompiler ${CMAKE_OSX_SYSROOT}")
```

Comment:
why is there the second Xcompiler argument: should it be:
``` cmake
-Xcompiler \"-isysroot ${CMAKE_OSX_SYSROOT}\"
```

Reply:
The way I have it is correct. If you put it in quotes, the space actually gets passed to the compiler.

Reply:
this is what you want, right? the argument to clang should be ``-isysroot /path/to/root``

Reply:
No. Clang needs two separate arguments, one `-isysroot`, one `/path/to/root`.